crates/bevy_render/src/view/window/mod.rs

```diff
@@ -304,19 +306,41 @@ pub fn prepare_windows(
                 })
         };
 
+        #[cfg(target_os = "linux")]
+        let is_nvidia = || {
+            render_instance
+                .enumerate_adapters(wgpu::Backends::VULKAN)
+                .iter()
+                .any(|adapter| adapter.get_info().name.starts_with("NVIDIA"))
```

Comment:
I'm not sure if `starts_with()` is good enough here. We've had a bug somewhere else because the vendor name wasn't at the start.

Reply:
In addition to the `name` field there seems to be `driver` that just says "NVIDIA" and nothing else. Would that be better? I don't have other cards to test with.

Reply:
Not sure how quirky these `AdapterInfo` strings can get but maybe just changing it to `name.to_uppercase().contains("NVIDIA")` could work. You could also add a fallback check of the `driver` string to account for cases in which the `name` would not match on an NVIDIA device, no idea if this would be necessary though. This is out of the scope of this PR but the same `starts_with()` method is currently used in defining `may_erroneously_timeout` in the lines above.

Reply:
I wouldn't add random extra checks if we don't know what edge case we are handling. This has worked fine for everyone who tested it.

Reply:
Checking for vendor id would probably be the best for reliability.

Reply:
I can confirm with a little test system that this spams `true`:

```rust
fn vendor_info_hex(adapter: Res<RenderAdapter>) {
    let nvidia = 0x10de;
    let vendor = adapter.get_info().vendor;

    info!("Is nvidia? {}", vendor == nvidia);
}
```

found the ID value here: https://envytools.readthedocs.io/en/latest/hw/pciid.html#introduction

Reply:
According to wgpu there might be other bytes non-zero: https://docs.rs/wgpu/latest/wgpu/struct.AdapterInfo.html#structfield.vendor

in the ID, so might need to mask off only the 16 bits we care about

```rust
let vendor = adapter.get_info().vendor & 0xFFFF;
```

Reply:
Thanks for investigating. I made this change. An enum based abstraction for the vendor check would probably be nice and should be used everywhere instead of strings.